force-app/main/default/lwc/nksSurvey/nksSurvey.js

```diff
@@ -0,0 +1,214 @@
+import { LightningElement, track, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import getSurvey from '@salesforce/apex/NKS_InternalSurveyController.getSurvey';
+import hasAnswered from '@salesforce/apex/NKS_InternalSurveyController.hasAnswered';
+import createAnsweredRecord from '@salesforce/apex/NKS_InternalSurveyController.createAnsweredRecord';
+import createResponseRecord from '@salesforce/apex/NKS_InternalSurveyController.createResponseRecord';
+export default class NksSurvey extends LightningElement {
+    surveyId;
+    backgroundColor;
+    title;
+    question;
+    isAnswered;
+    isRendered = false;
+    recordTypeId;
+    startDate;
+    endDate;
+
+    @track rating;
+    @track comment;
+    @track show = true;
+
+    @wire(getSurvey)
+    wiredSurvey({ data, error }) {
+        if (data) {
+            this.surveyId = data.Id;
+            this.backgroundColor = data.NKS_Background_Color__c;
+            this.title = data.NKS_Title__c;
+            this.question = data.NKS_Question__c;
+            this.startDate = data.NKS_Start_Date__c;
+            this.endDate = data.NKS_End_Date__c;
+        } else if (error) {
+            console.log('Problem getting survey: ' + error);
+        }
+    }
+
+    get background() {
+        return `background-color: ${this.backgroundColor}`;
+    }
+
+    get hide() {
+        return !this.show || this.isAnswered;
+    }
+
+    get isValid() {
+        let valid = false;
+        if (this.surveyId) {
+            if (this.startDate && this.endDate) {
+                valid = new Date(this.endDate) - new Date(this.startDate) > 0 ? true : false;
+            } else {
+                valid = true;
+            }
+        }
+        return valid;
+    }
+
+    renderedCallback() {
+        if (this.surveyId && !this.isRendered) {
+            hasAnswered({ surveyId: this.surveyId }).then((res) => {
+                this.isAnswered = res;
+                this.isRendered = true;
+                console.log('Survey is answered: ', res);
+            });
+        }
+    }
+
+    handleClick(event) {
+        this.rating = event.currentTarget.getAttribute('value');
+        const currentId = event.currentTarget.getAttribute('data-id');
+        this.handleSelected(currentId);
+    }
+
+    handleSelected(currentId) {
```

Comment:
Litt usikker på om det er bedre å ha det som static resources istedenfor, men det var vel noe som var litt problematisk, så vidt jeg husker?
Jeg ville uansett ha gjort dette i en for-loop istedenfor å ha så mange ifs. 
Noe a la:
```
const emojiIds = ['emoji1', 'emoji2', 'emoji3', 'emoji4', 'emoji5'];

    for (const emojiId of emojiIds) {
        const paths = this.template.querySelectorAll(`li[data-id="${emojiId}"] path`);

        if (currentId === emojiId) {
            paths[0].setAttribute('class', 'select1');
            paths[1].setAttribute('class', 'select2');
        } else {
            paths[0].setAttribute('class', 'path1');
            paths[1].setAttribute('class', 'path2');
        }
```

Reply:
stemmer, det var vanskelig å håndtere med bare en svg filte per emoji